src/Game-core/sound.cpp

```diff
@@ -0,0 +1,79 @@
+#include "sound.h"
+#include <iostream>
+
+
+
+sound::sound()
+{
+	isDeath = false;
+	isFood = false;
+	if (!eatSoundBuffer.loadFromFile(eatSoundPath))
+	{
+		std::cout << "hit sound file is either corrupted or location given is wrong" << std::endl;
+	}
+
+	if (!deathSoundBuffer.loadFromFile(deathSoundPath))
+	{
+		std::cout << "death sound file is either corrupted or location given is wrong" << std::endl;
+	}
+	
+	if (!bgm.openFromFile(BGMPath))
+	{
+		std::cout << "BGM sound file is either corrupted or location given is wrong" << std::endl;
+	}
+
+	if (!MM.openFromFile(menuMusicPath))
+	{
+		std::cout << "main menu sound file is either corrupted or location given is wrong" << std::endl;
+	}
+
+	eatSound.setBuffer(eatSoundBuffer);	
+	deathSound.setBuffer(deathSoundBuffer);
+	bgm.setVolume(60);
+	MM.setVolume(60);
+}
+
+
+void sound::onCollisionSound()
+{
+	if (isDeath)
+	{
+		deathSound.play();
+		while (deathSound.getStatus() == sf::Sound::Playing)
+		{
+			sf::sleep(sf::milliseconds(100));
+		}
+	}
+	if (isFood)
+	{
+		eatSound.play();
+	}
+	isDeath = false;
+	isFood = false;
+}
+
+void sound::BGM()
```

Comment:
`startBGM` might be a better name.

Reply:
Here i simply used BGM because this is doing both the tasks of starting and stopping the background music. And it is working in way that at every alternate calls it will start and stop the BGM

Reply:
This makes the actual function of method hard to understand when reading the code unless you open the function definition. Consider splitting it into `StartBGM` and `StopBGM`